src/Learning/KWDataPreparation/KWClassStats.cpp

```diff
@@ -828,23 +829,26 @@ void KWClassStats::WriteReport(ostream& ost)
 	    << "\t" << TSV::Export(GetClass()->GetName()) << "\n";
 
 	// Nombres d'attributs par type
-	ost << "Variables"
-	    << "\n";
-	nTotalAttributeNumber = 0;
-	for (nType = 0; nType < KWType::None; nType++)
+	if (IsStatsComputed())
 	{
-		if (KWType::IsData(nType))
+		ost << "Variables"
+		    << "\n";
+		nTotalAttributeNumber = 0;
+		for (nType = 0; nType < KWType::None; nType++)
 		{
-			nAttributeNumber = GetUsedAttributeNumberForType(nType);
-			nTotalAttributeNumber += nAttributeNumber;
-			if (nAttributeNumber > 0)
-				ost << "\t" << KWType::ToString(nType) << "\t" << nAttributeNumber << "\n";
+			if (KWType::IsData(nType))
+			{
+				nAttributeNumber = GetUsedAttributeNumberForType(nType);
+				nTotalAttributeNumber += nAttributeNumber;
```

Comment:
From a stylistic standpoint, I would have put the update of `nTotalAttributeNumber` after the conditional displaying of the value of the `nAttributeNumber` increment. Thus, the execution flow seems smoother to me:
1. we get `nAttributeNumber`
2. if > 0, we display it
3. we use it to increment `nTotalAttributeNumber` with it.

And after  looking at the execution flow in this manner, I would put step 3 (the update of `nTotalAttributeNumber` inside the `if` (just after the display of `nAttributeNumber`, so that it still is the _potentially_ last action of this block).

Reply:
Fait: l'incrémentation de nTotalAttributeNumber est désormais en fin de bloc

Reply:
I would only do the incrementation of `nTotalAttributeNumber` if `nAttributeNumber > 0`, that is, inside the `if` block.